lib/Doctrine/ORM/Configuration.php

```diff
@@ -349,11 +350,9 @@ public function addNamedNativeQuery($name, $sql, Query\ResultSetMapping $rsm)
      *
      * @param string $name The name of the query.
      *
-     * @psalm-return array{string, ResultSetMapping} A tuple with the first
-     *                                               element being the SQL
-     *                                               string and the second
-     *                                               element being the
-     *                                               ResultSetMapping.
+     * @return mixed[]
```

Comment:
Why not `ResultSetMapping[]`?

Reply:
From the psalm-return annotation the result is something like
```
[
    [
        'string'
        new ResultSetMapping()
    ]
]
```

It is not `array<string, ResultSetMapping>`

Reply:
`@return string[]|ResultSetMapping[]` would be closer from the truth while still compatible with IDEs I think

Reply:
This would be either an array of strings or of ResultSetMapping but not a complex array result.

Reply:
I've re-read the return type and it says:
> A tuple with the first element being the SQL string and the second element being the ResultSetMapping.

So probably the IDE-friendly type should be:
```suggestion
     * @return (string|ResultSetMapping)[]
```

Reply:
There is no simple phpdoc annotation to describe complex arrays. 
I don't think there is any different between the meaning of `(string|ResultSetMapping)[]` or `string[]|ResultSetMapping[]`. Both do not represent a complex array including a string AND a ResultSetMapping which means it opens room for wrong expectations.
As @orklah wrote as well, at least the psalm annotation is exact. This PR tries to support as many IDEs as possible with as precise typings as possible. In this particular case (from the function typing) before it wasn't even known that this function returns an array.

Reply:
> I don't think there is any different between the meaning of (string|ResultSetMapping)[] or string[]|ResultSetMapping[]

There is a difference, and an important one in this context: the second notation excludes the exact situation we're in, where we have an array with two different kind of elements; it's either all strings or all ResultSetMappings. And that's not our case.

If IDEs cannot decode the first notation, well, that's the only case for not using it.